src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalErrorTests.cs

```diff
@@ -516,6 +516,62 @@ public static int Main ()
             ParserErrorMessageTests.ParseAndValidate(test, Diagnostic(ErrorCode.ERR_OpenEndedComment, ""));
         }
 
+        [Fact]
+        public void CS1035ERR_OpenEndedComment_Razor()
+        {
+            var test = @"
+public class MainClass
+    {
+    public static int Main ()
+        {
+        return 1;
+        }
+    }
+//Comment lacks closing */
+@*    
+";
+
+            ParserErrorMessageTests.ParseAndValidate(test,
+                // (10,1): error CS1056: Unexpected character '@'
+                // @*    
+                Diagnostic(ErrorCode.ERR_UnexpectedCharacter, "@").WithArguments("@").WithLocation(10, 1),
+                // (10,1): error CS1035: End-of-file found, '*/' expected
```

Comment:
I did not feel it was worth the effort to customize the error message for `*@`, particularly since that's not actually valid syntax.

Reply:
This is my only nit, as from a C# user perspective it is confusing. Sure, it's not valid syntax but I never like it when error messages are misleading, especially for newer users who are more likely to make mistakes and less likely to be able to understand them. Given that we have the `Unexpected characted '@'` though, it's probably fine.